slm_porosity/heat_transfer.f90

```diff
@@ -1116,6 +1124,32 @@ PURE FUNCTION laser_distribution (x_, nlocal, factor)
     laser_distribution = EXP(-SUM(factor_*(x_-x_center)**2, 2))/pi**(.5*(dim-1))
   END FUNCTION laser_distribution
 
+  FUNCTION laser_heat_source (x_, nlocal)
+    IMPLICIT NONE
+    REAL(pr), INTENT(IN) :: x_(nlocal,dim)
+    INTEGER,  INTENT(IN) :: nlocal
+    REAL(pr) :: laser_heat_source(nlocal)
+    REAL(pr) :: halfwidth(dim), position(dim)
+    INTEGER  :: i
+
+    position = laser_position(t)
+    halfwidth = 1.0_pr
+    halfwidth(dim) = optical_penetration_depth
+    laser_heat_source = 2*laser_heat_flux(psi_prev)
```

Comment:
`laser_heat_flux` can be excluded from this function since they are independent.